Mathlib/Data/List/Join.lean

```diff
@@ -52,7 +52,7 @@ theorem join_filter_isEmpty_eq_false [DecidablePred fun l : List α => l.isEmpty
   | [] :: L => by
       simp [join_filter_isEmpty_eq_false (L := L), isEmpty_iff_eq_nil]
   | (a :: l) :: L => by
-      simp [join_filter_isEmpty_eq_false (L := L), isEmpty_cons]
+      simp [join_filter_isEmpty_eq_false (L := L)]
```

Comment:
Is this some merge weirdness, or genuinely intended as part of this patch?

Reply:
I'm not sure. This is what is currently on `nightly-testing`: it doesn't seem to be on @Vtec234's `lean-pr-testing-2964` branch...

Given it's a good change I'm inclined to just sneak it in here?